test/remain.c

```diff
@@ -138,11 +140,81 @@ static int test_remain_thread(void)
 }
 
 
+static void mqueue_handler(int id, void *dat, void *arg)
+{
+	struct data *data = arg;
+	(void)id;
+	(void)dat;
+
+	++data->mqueue_called;
+
+	/* Stop re_main loop */
+	re_cancel();
+}
+
+
+static int enterleave_thread_handler(void *arg)
+{
+	struct data *data = arg;
+
+	/* Enter an 're' thread */
+	re_thread_enter();
+
+	/* note: allocated from this thread */
+	int err = mqueue_alloc(&data->mq, mqueue_handler, data);
```

Comment:
If this returns ENOMEM in OOM memory test `re_cancel` is not called and re_poll timeouts, I think.

Reply:
Maybe starting a fallback timer could help. Or ignoring `re_main_timeout` error.